providers/Spotify/mod.ts

```diff
@@ -0,0 +1,321 @@
+			this.constructReleaseApiUrl(),
+			this.options.snapshotMaxTimestamp,
+		);
+		const release = cacheEntry.content;
+
+		this.updateCacheTime(cacheEntry.timestamp);
+		return release;
+	}
+
+	private async getRawTracklist(rawRelease: Album): Promise<Track[]> {
+		const allTracks: SimplifiedTrack[] = [...rawRelease.tracks.items];
+
+		// The initial response contains max. 50 tracks. Fetch the remaining
+		// tracks with separate requests if needed.
+		let nextUrl = rawRelease.tracks.next;
+		while (nextUrl && allTracks.length < rawRelease.tracks.total) {
+			const cacheEntry = await this.provider.query<ResultList<SimplifiedTrack>>(
+				new URL(nextUrl),
+				this.options.snapshotMaxTimestamp,
+			);
+			this.updateCacheTime(cacheEntry.timestamp);
+			allTracks.push(...cacheEntry.content.items);
+			nextUrl = cacheEntry.content.next;
+		}
+
+		// Load full details including ISRCs
+		if (this.options.withISRC) {
+			return this.getRawTrackDetails(allTracks);
+		} else {
+			return allTracks.map((track) => track as Track);
+		}
+	}
+
+	private async getRawTrackDetails(simplifiedTracks: SimplifiedTrack[]): Promise<Track[]> {
+		const allTracks: Track[] = [];
+		const trackIds = simplifiedTracks.map((track) => track.id);
+
+		// The SimplifiedTrack entries do not contain ISRCs.
+		// Perform track queries to obtain the full details of all tracks.
+		// Each query can return up to 50 tracks.
+		const maxResults = 50;
+		const apiUrl = new URL('tracks', this.provider.apiBaseUrl);
+		for (let index = 0; index < trackIds.length; index += maxResults) {
+			apiUrl.searchParams.set('ids', trackIds.slice(index, index + maxResults).join(','));
+			apiUrl.search = apiUrl.searchParams.toString();
+			const cacheEntry = await this.provider.query<TrackList>(
+				apiUrl,
+				this.options.snapshotMaxTimestamp,
+			);
+			this.updateCacheTime(cacheEntry.timestamp);
+			allTracks.push(...cacheEntry.content.tracks);
+		}
+
+		return allTracks;
+	}
+
+	protected async convertRawRelease(rawRelease: Album): Promise<HarmonyRelease> {
+		this.id = rawRelease.id;
+		const rawTracklist = await this.getRawTracklist(rawRelease);
+		const media = this.convertRawTracklist(rawTracklist);
+		const artwork = selectLargestImage(rawRelease.images, ['front']);
+		return {
+			title: rawRelease.name,
+			artists: rawRelease.artists.map(this.convertRawArtist.bind(this)),
+			gtin: rawRelease.external_ids.ean || rawRelease.external_ids.upc,
+			externalLinks: [{
+				url: new URL(rawRelease.external_urls.spotify),
+				types: ['free streaming'],
+			}],
+			media,
+			releaseDate: parseHyphenatedDate(rawRelease.release_date),
+			copyright: this.getCopyright(rawRelease.copyrights),
+			status: 'Official',
+			packaging: 'None',
+			images: artwork ? [artwork] : [],
+			labels: splitLabels(rawRelease.label),
+			availableIn: rawRelease.available_markets,
+			info: this.generateReleaseInfo(),
+		};
+	}
+
+	private convertRawTracklist(tracklist: Track[]): HarmonyMedium[] {
```

Comment:
If we change the return type of `getRawTracklist`:
```suggestion
	private convertRawTracklist(tracklist: SimplifiedTrack[]): HarmonyMedium[] {
```